### PR TITLE
workflows: various fixes & consistency passes

### PR DESCRIPTION
**Pull request by nbusseneau**
Please see individual commits :)

This supersedes the original #16595.

### CODE REVIEW

**Comment by nbusseneau** [comment]
It seems we still have issues with the multicluster workflow: https://github.com/cilium/cilium/actions/runs/1001641767
```
📋 Test Report
❌ 2/9 tests failed (12/89 actions), 0 warnings, 0 tests skipped, 2 scenarios skipped:
Test [no-policies]:
  ❌ no-policies/pod-to-remote-nodeport/curl-0: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-other-node (echo-other-node:8080)
  ❌ no-policies/pod-to-remote-nodeport/curl-4: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ no-policies/pod-to-remote-nodeport/curl-5: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ no-policies/pod-to-remote-nodeport/curl-6: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-other-node (echo-other-node:8080)
  ❌ no-policies/pod-to-remote-nodeport/curl-10: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ no-policies/pod-to-remote-nodeport/curl-11: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
Test [allow-all]:
  ❌ allow-all/pod-to-remote-nodeport/curl-2: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-other-node (echo-other-node:8080)
  ❌ allow-all/pod-to-remote-nodeport/curl-4: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
Error: Connectivity test failed: 2 tests failed
  ❌ allow-all/pod-to-remote-nodeport/curl-5: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ allow-all/pod-to-remote-nodeport/curl-8: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-other-node (echo-other-node:8080)
  ❌ allow-all/pod-to-remote-nodeport/curl-10: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ allow-all/pod-to-remote-nodeport/curl-11: cilium-test/client2-666976c95b-blprg (10.76.1.228) -> cilium-test/echo-same-node (echo-same-node:8080)
Error: Process completed with exit code 1.
```
However, as @tklauser was saying offline:
> At least it's only 2/9 tests failed compared to 9/9 tests failed before :smile:

Soooo I guess it's progress...?

**Review by tklauser — APPROVED** [review]

**Comment by nbusseneau** [comment]
Links to working runs:
- AKS https://github.com/cilium/cilium/actions/runs/1004803015
- EKS https://github.com/cilium/cilium/actions/runs/1004803009
- AWS-CNI https://github.com/cilium/cilium/actions/runs/1004803008
- GKE https://github.com/cilium/cilium/actions/runs/1004803017
- Multicluster https://github.com/cilium/cilium/actions/runs/1004803004

Multicluster testing will now be officially fixed 🎉

**Review by michi-covalent — APPROVED** [review]

File: .github/workflows/conformance-aks.yaml
Comment (pchaigno) [inline .github/workflows/conformance-aks.yaml:249]: Maybe worth pulling in https://github.com/cilium/cilium-cli/pull/390 at the same time?

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/conformance-aks.yaml
Comment (michi-covalent) [reply .github/workflows/conformance-aks.yaml:249]: looks like it's blocked by this flake: https://github.com/cilium/cilium-cli/issues/399 😢

Comment (pchaigno) [reply .github/workflows/conformance-aks.yaml:249]: Yes, but we don't need to wait for that commit to be merged in cilium-cli to pull it in cilium/cilium :-)

Comment (nbusseneau) [reply .github/workflows/conformance-aks.yaml:249]: Thanks, added in 086ff29af18d85c02848a2dced47683cec3cca97.

**Comment by tklauser** [comment]
Received reviews from all requested teams and all GH actions passed. Marking as ready to merge to unblock PRs stuck on the failing multicluster test.

**Pull request merged**
